pkg/csi/service/common/authmanager.go

```diff
@@ -325,17 +351,20 @@ func getDatastoresWithBlockVolumePrivs(ctx context.Context, vc *vsphere.VirtualC
 	return dsURLToInfoMap, nil
 }
 
-// Creates a map of vsan datastores to file service status (enabled/disabled).
-func getDsToFileServiceEnabledMap(ctx context.Context, vc *vsphere.VirtualCenter) (map[string]bool, error) {
+// Returns
+// i. a map of vsan datastores to file service status (enabled/disabled).
+// ii. a map of cluster ID to file service enabled vsan datastore URLs.
+func createFSEnabledDsMapsForVC(ctx context.Context, vc *vsphere.VirtualCenter) (map[string]bool, map[string][]string, error) {
```

Comment:
function name doesn't look good based on previous comment.

Reply:
But now we're returning multiple maps, and this takes into account all datastores for the VC.
What do you think can be an appropriate name?

Reply:
`createFSEnabledDsMapsForVC` returning both maps in this function doesn't make sense.

Can you split it into 2 different functions. what we need here is clusters which have FS enabled and datastores within those clusters.

First function is `getFSEnabledClusters`
second function is `getDatastores([]clusters)`

createFSEnabledDsMapsForVC doesn't give the right sense.